GestureHandlerModule.js

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import { TouchableWithoutFeedback, View } from 'react-native';
+
+import State from './State';
+
+class BaseGestureHandler extends React.Component {
+  clearState = () => {};
+
+  handleHandlerStateChange(event, oldState, state) {
+    const { enabled, onHandlerStateChange } = this.props;
+
+    if (enabled !== false && onHandlerStateChange) {
+      const {
+        nativeEvent: {
+          locationX: x,
+          locationY: y,
+          pageX: absoluteX,
+          pageY: absoluteY,
+        },
+      } = event;
+
+      onHandlerStateChange({
+        nativeEvent: {
+          oldState,
+          state,
+          x,
+          y,
+          absoluteX,
+          absoluteY,
+          pointerInside: true,
+        },
+      });
+    }
+  }
+
+  handleFailed = event => {
+    this.clearState();
+    this.handleHandlerStateChange(event, State.ACTIVE, State.FAILED);
+  };
+
+  handleEnd = event => {
+    this.clearState();
+    this.handleHandlerStateChange(event, State.ACTIVE, State.END);
+  };
+
+  handleActivate = event => {
+    this.handleHandlerStateChange(event, State.BEGAN, State.ACTIVE);
+  };
+
+  handleBegan = event => {
+    this.handleHandlerStateChange(event, State.UNDETERMINED, State.BEGAN);
+  };
+}
+
+class UnimplementedGestureHandler extends BaseGestureHandler {
+  render() {
+    return this.props.children;
+  }
+}
+
+const handlers = {
+  NativeViewGestureHandler: class NativeViewGestureHandler extends BaseGestureHandler {
+    render() {
+      const { children } = this.props;
+
+      return children;
+    }
+  },
+
+  TapGestureHandler: class TapGestureHandler extends BaseGestureHandler {
```

Comment:
I'm not really into using inheritance in that way. I think it's not a common practice to use it when in comes to react component. It's not very readable then and it's not very popular for these cases. Would you mind rather using a composition here in similar way we're doing it in `GestureHandler.js`?

Reply:
Just changed it 😊